documentation/design/proposals/shared-infrastructure.adoc

```diff
@@ -12,23 +12,33 @@ This document will consider the alternative with a new address space type + an a
 
-AddressSpace resource will be deprecated, and a restriction of 1 instance of messaging per Kubernetes namespace is mandated. New CRDs for configuring messaging endpoints, connectors and plan/authservice config is defined. The AddressSpace resource is replaced by `MessagingConfig`, `MessagingEndpoint` and `MessagingConnector` resources.
+* `MessagingPlan`: Conceptually replaces the `AddressSpacePlan`, but can be optional or enforced. Is not referencing `AddressPlan`s.
 
-The `MessagingConfig` type, which will be a singleton per namespace, is introduced to contain the messaging configuration of a namespace.
+* `AddressPlan`: Defines resources requested for an address using this plan. Referencing an AddressPlan may be optional but can be enforced. Resources are specified in terms of queue memory usage and depth, number of paritions, and type-specific settings such as DLQ and expiry queue.
 
-The `Address` and `AddressPlan` types will remain, but the schema will be extended to support options required by the shared infrastructure.
+* `AuthenticationService`: Gaining ability to use selectors to configure to which namespaces it should be applied.
 
-The `qdr-operator` and `activemq-artemis-operator` operators will be used for managing routers and brokers.
 
-The `cert-manager` operator will be used to manage certificates, both internal and external.
+At the tenant level, the API will change to simplify the user experience:
 
-The `MessagingPlan`, `AddressPlan` and `AuthenticationService` resources may additionally be referenced by their namespace. This allows having a service operator persona that does not manage or have access to the messaging infrastructure, but manages the plans and authentication service configurations offered to messaging tenants.
+A restriction of 1 instance of messaging per Kubernetes namespace is mandated. New CRDs for configuring messaging endpoints, connectors and plan/authservice config is defined. The AddressSpace resource is replaced by `MessagingTenant`, `MessagingEndpoint` and `MessagingConnector` resources:
+
+* `MessagingTenant`: a singleton per namespace, is introduced to contain the messaging configuration of a namespace. Auto-created when initial `Address` is created. May be used to override the default `MessagingPlan` and `AuthenticationService` of a tenant.
+
+* `MessagingConnector`: similar to existing connector spec in AddressSpace. 1 instance configures 1 connector.
+
+* `MessagingEndpoint`: allows configuring endpoints to access messaging infrastructure. Cluster-local endpoint is created by default on initial `Address` creation.
 
-MessagingUser and Address will no longer needed to be prefixed with address space name!
+* `Address`: similar to existing type. Name is no longer prefixed, `.spec.address` is optional. `.spec.plan` is optional if permitted by plan config. Addition per-type sub-objects such as `.spec.queue` or `.spec.topic` may be specified to override plan properties (if allowed by plan).
```

Comment:
by "if permitted by plan config" it's referring to MessagingPlan right?

Reply:
It is referring to the AddressPlan which covers per-address limits. The MessagingPlan covers per-namespace limits spanning all addresses in a namespace. I will make the sentence clearer, thanks!

Reply:
but how can be the .spec.plan of an address be optional or not if permitted by AddressPlan config, that was my point

Reply:
It is optional in the sense that the tenant does not have to specify it. However, if there is an addressplan with a selector for the namespace that address is created in, that plan will be the “default” if no plan is set. 

If there are multiple addressplans , we can select the default based on some annotation (similar to storage class for kubernetes). We can also define that if there is a plan with matching selector but without any annotation indicating it should be default, we require the user to specify the plan.

Reply:
I got it now, thanks Ulf :)